lib/acme/client/jwk/base.rb

```diff
@@ -0,0 +1,77 @@
+class Acme::Client::JWK::Base
```

Comment:
That matches the subclasses defined in this repo, but isn't very general. For example, the JWK that I'm using in my project has a signature like

```ruby
class HsmBackedJwk < Acme::Client::JWK::Base
  def initialize(hsm_key_label)
    ...
  end
end
```

I don't feel strongly though since subclasses will be overriding the method anyway. I'm happy to add it with the `_private_key` argument.